programming-extensions/programming-extension-vfsprovider/src/main/java/org/objectweb/proactive/extensions/vfsprovider/client/ProActiveFileObject.java

```diff
@@ -737,4 +800,125 @@ public void writeUTF(String str) throws IOException {
             getDataOutputStream().writeUTF(str);
         }
     }
+
+    static final class DefaultFileSelectorInfo implements FileSelectInfo {
+        private FileObject baseFolder;
+
+        private FileObject file;
+
+        private int depth;
+
+        @Override
+        public FileObject getBaseFolder() {
+            return baseFolder;
+        }
+
+        public void setBaseFolder(final FileObject baseFolder) {
+            this.baseFolder = baseFolder;
+        }
+
+        @Override
+        public FileObject getFile() {
+            return file;
+        }
+
+        public void setFile(final FileObject file) {
+            this.file = file;
+        }
+
+        @Override
+        public int getDepth() {
+            return depth;
+        }
+
+        public void setDepth(final int depth) {
+            this.depth = depth;
+        }
+    }
+
+    private static int getConfiguredNumberOfThreads() {
+        if (CentralPAPropertyRepository.PA_VFSPROVIDER_CLIENT_FIND_FILES_THREAD_NUMBER.isSet()) {
+            try {
+                return CentralPAPropertyRepository.PA_VFSPROVIDER_CLIENT_FIND_FILES_THREAD_NUMBER.getValue();
+            } catch (Exception e) {
+                logger.error("Invalid value for " +
+                             CentralPAPropertyRepository.PA_VFSPROVIDER_CLIENT_FIND_FILES_THREAD_NUMBER.getName(), e);
+                return Runtime.getRuntime().availableProcessors() * 2 - 1;
+            }
+        } else {
+            return Runtime.getRuntime().availableProcessors() * 2 - 1;
```

Comment:
can you merge the two `return Runtime.getRuntime().availableProcessors() * 2 - 1;` into one?